app/src/main/shared/components/OneTimeButton.tsx

```diff
@@ -8,16 +8,17 @@ interface Props {
     token: string
     enabled?: boolean;
     refreshToken: () => void;
+    onClickOuterEvent?: () =>void;
```

Comment:
I think this could just be called `onClick`. Better to have it easy to understand for components using the `OneTimeButton`. If you'd like to keep things unambiguous, maybe rename the internal method to something like `internalOnClickHandler`.